seqr/views/utils/anvil_metadata_utils.py

```diff
@@ -536,3 +542,28 @@ def _get_parsed_saved_discovery_variants_by_family(families):
     return get_saved_discovery_variants_by_family(
         {'family__id__in': families}, _format_variants, lambda saved_variant: saved_variant.pop('familyId'),
     )
+
+
+def get_omim_conditions_by_id_gene(mim_numbers):
+    omim_conditions_by_id_gene = defaultdict(lambda: defaultdict(list))
+    for omim in Omim.objects.filter(phenotype_mim_number__in=mim_numbers).values(
+        'phenotype_mim_number', 'phenotype_description', 'phenotype_inheritance', 'chrom', 'start', 'end', 'gene__gene_id',
+    ):
+        omim_conditions_by_id_gene[omim['phenotype_mim_number']][omim['gene__gene_id']].append(omim)
+    return omim_conditions_by_id_gene
+
+
+def variant_omim_conditions(conditions_by_gene, variants, get_variant_genes):
+    gene_ids = set()
+    positions = set()
+    for variant in variants:
+        gene_ids.update(get_variant_genes(variant))
+        positions.add((variant['chrom'], variant['pos']))
+
+    conditions = [
+        c for c in conditions_by_gene[None]
+        if any(c['chrom'] == chrom and c['start'] <= pos <= c['end'] for chrom, pos in positions)
+    ]
+    for gene_id in gene_ids:
+        conditions += conditions_by_gene[gene_id]
```

Comment:
are we ok with there potentially being duplicate conditions here?

Reply:
good catch!

Reply:
this got refactored to prevent merge conflicts with a different open PR, but the latest updates fix the issue because conditions are put into a set before being formatted and returned so duplicates are okay: https://github.com/broadinstitute/seqr/pull/3799/files#diff-fcba4e6ac6983d5e2f987e729d8b190d3add7e8dd3b75a59b9c274bbdadd7ce6R558